Update information for ROCm 5.4 for MIGraphX and ROCm Builds

TedThemistokleous opened the pull request:
-Add MIGraphX-EP page for install/example
-Update version info for ROCm version for both MIGraphx and ROCm EPs to 5.4 -Update hooks to the latest ROCm Pytorch supported (1.12.1 -> 1.13) -Remove (Preview) from MIGraphx and ROCm EP notes
-Update ROCm & MIGraphX EP.md files with ROCm version and pytorch links

### Description
<!-- Describe your changes. -->
Update documentation about ROCm and MIGraphx with newest ROCm 5.4 stack


### Motivation and Context
<!-- - Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here. -->

Update things for whats able to be supported.

TedThemistokleous commented:
@cloudhan  this is important since a lot of version info is outdated on these pages. Feel free to let us know if there's anything else required for these changes.

PeixuanZuo commented on docs/execution-providers/ROCm-ExecutionProvider.md:
```diff
@@ -19,14 +19,20 @@ The ROCm Execution Provider enables hardware accelerated computation on AMD ROCm
-|main|5.2.3|
+|main|5.4|
+|1.13|5.4|
+|1.13|5.3|
```
Becuase of rocblas bug on ROCm5.3, we skip ROCm5.3 and use ROCm5.3.2 now. Could you add the minor version here?

TedThemistokleous replied:
Yup no problem I can update this

TedThemistokleous replied:
Updated

PeixuanZuo commented on docs/install/index.md:
```diff
@@ -251,5 +251,6 @@ The _location_ needs to be specified for any specific version other than the def
 |[*Preview*] PyTorch 1.11 (ROCm 5.1.1)|[**onnxruntime_stable_torch1110.rocm511**](https://download.onnxruntime.ai/onnxruntime_stable_rocm511.html)|[onnxruntime_nightly_torch1110.rocm511](https://download.onnxruntime.ai/onnxruntime_nightly_rocm511.html)|
-|[*Preview*] PyTorch 1.11 (ROCm 5.2)||[onnxruntime_nightly_torch1110.rocm52](https://download.onnxruntime.ai/onnxruntime_nightly_rocm511.html)|
-|[*Preview*] PyTorch 1.12.1 (ROCm 5.2.3)||[onnxruntime_nightly_torch1121.rocm523](https://download.onnxruntime.ai/onnxruntime_nightly_rocm523.html)|
+|PyTorch 1.11 (ROCm 5.2)||[onnxruntime_nightly_torch1110.rocm52](https://download.onnxruntime.ai/onnxruntime_nightly_rocm511.html)|
```
torch1.11 + ROCm52 is stop nightly building now. Could you please use https://download.onnxruntime.ai/onnxruntime_stable_rocm52.html with tag [onnxruntime_stable_torch1110.rocm52] here?

TedThemistokleous replied:
Yes I can update this

TedThemistokleous replied:
done

PeixuanZuo commented on docs/install/index.md:
```diff
@@ -251,5 +251,6 @@ The _location_ needs to be specified for any specific version other than the def
-|[*Preview*] PyTorch 1.12.1 (ROCm 5.2.3)||[onnxruntime_nightly_torch1121.rocm523](https://download.onnxruntime.ai/onnxruntime_nightly_rocm523.html)|
+|PyTorch 1.11 (ROCm 5.2)||[onnxruntime_nightly_torch1110.rocm52](https://download.onnxruntime.ai/onnxruntime_nightly_rocm511.html)|
+|PyTorch 1.12.1 (ROCm 5.2.3)||[onnxruntime_nightly_torch1121.rocm523](https://download.onnxruntime.ai/onnxruntime_nightly_rocm523.html)|
+|PyTorch 1.13 (ROCm 5.2.3)||[onnxruntime_nightly_torch1121.rocm523](https://download.onnxruntime.ai/onnxruntime_nightly_rocm523.html)|
```
We have https://download.onnxruntime.ai/onnxruntime_nightly_rocm532.html with torch1.12.1 now. Could you please add it?

TedThemistokleous replied:
Will do!

TedThemistokleous replied:
Done

PeixuanZuo replied:
We have whl with ROCm5.4(https://download.onnxruntime.ai/onnxruntime_nightly_rocm54.html) now, Could you please add it? Thanks!

TedThemistokleous replied:
Done! Let me know if there's anything else

PeixuanZuo commented:
We have some other "Preview" on https://onnxruntime.ai/, the file is https://github.com/microsoft/onnxruntime/blob/gh-pages/index.html#L243, Could you please remove it on this PR?

ytaous commented:
Can u please add preview link like so? https://github.com/microsoft/onnxruntime/pull/12971
Also, this would be good ref for where "other Preview" might be, thx.

PeixuanZuo commented:
Because of the issue(https://github.com/microsoft/onnxruntime/issues/13676), we don't have ROCm5.4 python package yet. It's better to merge this PR after we release ROCm5.4 nightly build python package. Thanks.

TedThemistokleous commented:
@ytaous I think the ones for MIGraphX updates are #13564 or #9317

@PeixuanZuo Sounds good. Please let me know if there's anything else as well on this PR we should change before #13676 goes in.

ytaous commented:
> @ytaous I think the ones for MIGraphX updates are #13564 or #9317
> 
> @PeixuanZuo Sounds good. Please let me know if there's anything else as well on this PR we should change before #13676 goes in.

Thanks, did u mean the staged view over here? https://faxu.github.io/onnxruntime/docs/ ? If so, as @PeixuanZuo pointed out, you will see "Preview" still there in some places. Might be good to do global search on the cloned repo as needed.
thx
![image](https://user-images.githubusercontent.com/4484531/208216632-04de4b13-c6f7-4d0a-b9cc-04a7febd07ca.png)

TedThemistokleous commented:
> > @ytaous I think the ones for MIGraphX updates are #13564 or #9317
> > @PeixuanZuo Sounds good. Please let me know if there's anything else as well on this PR we should change before #13676 goes in.
> 
> Thanks, did u mean the staged view over here? https://faxu.github.io/onnxruntime/docs/ ? If so, as @PeixuanZuo pointed out, you will see "Preview" still there in some places. Might be good to do global search on the cloned repo as needed. thx ![image](https://user-images.githubusercontent.com/4484531/208216632-04de4b13-c6f7-4d0a-b9cc-04a7febd07ca.png)

@ytaous  Saw the extra "Preview" tags and just removed them when doing a find replace in install/index.md. 

@PeixuanZuo , I've also added the link for ROCm 5.4 here as well. 

Please let me know if there is any other changes needed.

ytaous commented on docs/build/eps.md:
```diff
@@ -634,8 +634,8 @@ See more information on the MIGraphX Execution Provider [here](../execution-prov
-* Install [ROCM](https://rocmdocs.amd.com/en/latest/Installation_Guide/Installation-Guide.html)
-  * The MIGraphX execution provider for ONNX Runtime is built and tested with ROCM3.3
+* Install [ROCm](https://docs.amd.com/bundle/ROCm-Installation-Guide-v5.4/page/How_to_Install_ROCm.html#_How_to_Install)
+  * The MIGraphX execution provider for ONNX Runtime is built and tested with ROCM5.4
```
>M [](http://example.com/codeflow?start=81&length=1)

for consistency "ROCm"

TedThemistokleous replied:
On it

TedThemistokleous replied:
done

ytaous commented on docs/build/eps.md:
```diff
@@ -634,8 +634,8 @@ See more information on the MIGraphX Execution Provider [here](../execution-prov
-* Install [ROCM](https://rocmdocs.amd.com/en/latest/Installation_Guide/Installation-Guide.html)
-  * The MIGraphX execution provider for ONNX Runtime is built and tested with ROCM3.3
+* Install [ROCm](https://docs.amd.com/bundle/ROCm-Installation-Guide-v5.4/page/How_to_Install_ROCm.html#_How_to_Install)
+  * The MIGraphX execution provider for ONNX Runtime is built and tested with ROCM5.4
```
>MIGraphX [](http://example.com/codeflow?start=8&length=8)

ROCm?

TedThemistokleous replied:
This is under the MIGraphx Section I believe.

ytaous commented on docs/execution-providers/index.md:
```diff
@@ -31,8 +31,8 @@ ONNX Runtime supports many different execution providers today. Some of the EPs
 |[TVM](../execution-providers/community-maintained/TVM-ExecutionProvider.md) (*preview*)|[DirectML](../execution-providers/DirectML-ExecutionProvider.md)|[Android Neural Networks API](../execution-providers/NNAPI-ExecutionProvider.md)|[Huawei CANN](../execution-providers/community-maintained/CANN-ExecutionProvider.md) (*preview*)|
-|[Intel OpenVINO](../execution-providers/OpenVINO-ExecutionProvider.md)|[AMD MIGraphX](../execution-providers/community-maintained/MIGraphX-ExecutionProvider.md) (*preview*)|[ARM-NN](../execution-providers/community-maintained/ArmNN-ExecutionProvider.md) (*preview*)|
-||[AMD ROCm](../execution-providers/ROCm-ExecutionProvider.md) (*preview*)|[CoreML](../execution-providers/CoreML-ExecutionProvider.md) (*preview*)|
+|[Intel OpenVINO](../execution-providers/OpenVINO-ExecutionProvider.md)|[AMD MIGraphX](../execution-providers/community-maintained/MIGraphX-ExecutionProvider.md) |[ARM-NN](../execution-providers/community-maintained/ArmNN-ExecutionProvider.md) (*preview*)|
```
>  [](http://example.com/codeflow?start=161&length=1)

space, but if it doesn't bother, u can ignore

TedThemistokleous replied:
Sure I can fix.

TedThemistokleous replied:
Done

ytaous commented on docs/execution-providers/index.md:
```diff
@@ -31,8 +31,8 @@ ONNX Runtime supports many different execution providers today. Some of the EPs
-|[Intel OpenVINO](../execution-providers/OpenVINO-ExecutionProvider.md)|[AMD MIGraphX](../execution-providers/community-maintained/MIGraphX-ExecutionProvider.md) (*preview*)|[ARM-NN](../execution-providers/community-maintained/ArmNN-ExecutionProvider.md) (*preview*)|
-||[AMD ROCm](../execution-providers/ROCm-ExecutionProvider.md) (*preview*)|[CoreML](../execution-providers/CoreML-ExecutionProvider.md) (*preview*)|
+|[Intel OpenVINO](../execution-providers/OpenVINO-ExecutionProvider.md)|[AMD MIGraphX](../execution-providers/community-maintained/MIGraphX-ExecutionProvider.md) |[ARM-NN](../execution-providers/community-maintained/ArmNN-ExecutionProvider.md) (*preview*)|
+||[AMD ROCm](../execution-providers/ROCm-ExecutionProvider.md) |[CoreML](../execution-providers/CoreML-ExecutionProvider.md) (*preview*)|
```
>  [](http://example.com/codeflow?start=62&length=1)

same - space

TedThemistokleous replied:
Done

ytaous commented on docs/install/index.md:
```diff
@@ -248,8 +248,11 @@ The _location_ needs to be specified for any specific version other than the def
+|PyTorch 1.8.1 (ROCm 4.2)|[**onnxruntime_stable_torch181.rocm42**](https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_stable_torch181.rocm42.html)|[onnxruntime_nightly_torch181.rocm42](https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_nightly_torch181.rocm42.html)|
+|PyTorch 1.9 (ROCm 4.2)|[**onnxruntime_stable_torch190.rocm42**](https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_stable_torch190.rocm42.html)|[onnxruntime_nightly_torch190.rocm42](https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_nightly_torch190.rocm42.html)|
+|PyTorch 1.11 (ROCm 5.1.1)|[**onnxruntime_stable_torch1110.rocm511**](https://download.onnxruntime.ai/onnxruntime_stable_rocm511.html)|[onnxruntime_nightly_torch1110.rocm511](https://download.onnxruntime.ai/onnxruntime_nightly_rocm511.html)|
+|PyTorch 1.11 (ROCm 5.2)||[onnxruntime_nightly_torch1110.rocm52](https://download.onnxruntime.ai/onnxruntime_stable_rocm52.html)|
```
>https://download.onnxruntime.ai/onnxruntime_stable_rocm52.html [](http://example.com/codeflow?start=65&length=62)

pls switch the order, 532 should be right above 54

ytaous commented on docs/install/index.md:
```diff
@@ -248,8 +248,11 @@ The _location_ needs to be specified for any specific version other than the def
+|PyTorch 1.11 (ROCm 5.1.1)|[**onnxruntime_stable_torch1110.rocm511**](https://download.onnxruntime.ai/onnxruntime_stable_rocm511.html)|[onnxruntime_nightly_torch1110.rocm511](https://download.onnxruntime.ai/onnxruntime_nightly_rocm511.html)|
+|PyTorch 1.11 (ROCm 5.2)||[onnxruntime_nightly_torch1110.rocm52](https://download.onnxruntime.ai/onnxruntime_stable_rocm52.html)|
+|PyTorch 1.12.1 (ROCm 5.2.3)||[onnxruntime_nightly_torch1121.rocm523](https://download.onnxruntime.ai/onnxruntime_nightly_rocm523.html)|
+|PyTorch 1.12.1 (ROCm 5.3.2)||[onnxruntime_nightly_torch1121.rocm532](https://download.onnxruntime.ai/onnxruntime_nightly_rocm532.html)|
```
pls switch the order, 532 should be right above 54

ytaous commented on docs/install/index.md:
```diff
@@ -248,8 +248,11 @@ The _location_ needs to be specified for any specific version other than the def
-|[*Preview*] PyTorch 1.12.1 (ROCm 5.2.3)||[onnxruntime_nightly_torch1121.rocm523](https://download.onnxruntime.ai/onnxruntime_nightly_rocm523.html)|
+|PyTorch 1.8.1 (ROCm 4.2)|[**onnxruntime_stable_torch181.rocm42**](https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_stable_torch181.rocm42.html)|[onnxruntime_nightly_torch181.rocm42](https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_nightly_torch181.rocm42.html)|
+|PyTorch 1.9 (ROCm 4.2)|[**onnxruntime_stable_torch190.rocm42**](https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_stable_torch190.rocm42.html)|[onnxruntime_nightly_torch190.rocm42](https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_nightly_torch190.rocm42.html)|
+|PyTorch 1.11 (ROCm 5.1.1)|[**onnxruntime_stable_torch1110.rocm511**](https://download.onnxruntime.ai/onnxruntime_stable_rocm511.html)|[onnxruntime_nightly_torch1110.rocm511](https://download.onnxruntime.ai/onnxruntime_nightly_rocm511.html)|
```
I think we can remove all deprecated ones, this and all above

TedThemistokleous replied:
Are we not concerned about compatibility? I can remove this but is the intent then, anything less than 5.2 will not be supported?

ytaous replied:
we can keep 5.1.1. 
@PeixuanZuo for second opinion.

PeixuanZuo replied:
We can keep 5.1.1, we still have active [onnxruntime_stable_rocm511](https://download.onnxruntime.ai/onnxruntime_stable_rocm511.htmll) now. we can remove it after it move to deprecated stable packages list.

PeixuanZuo replied:
But we will release ort 1.14 soon and the 511 is move to deprecated soon, so remove it now is ok.

ytaous commented on docs/execution-providers/index.md:
```diff
@@ -31,8 +31,8 @@ ONNX Runtime supports many different execution providers today. Some of the EPs
 |[TVM](../execution-providers/community-maintained/TVM-ExecutionProvider.md) (*preview*)|[DirectML](../execution-providers/DirectML-ExecutionProvider.md)|[Android Neural Networks API](../execution-providers/NNAPI-ExecutionProvider.md)|[Huawei CANN](../execution-providers/community-maintained/CANN-ExecutionProvider.md) (*preview*)|
-|[Intel OpenVINO](../execution-providers/OpenVINO-ExecutionProvider.md)|[AMD MIGraphX](../execution-providers/community-maintained/MIGraphX-ExecutionProvider.md) (*preview*)|[ARM-NN](../execution-providers/community-maintained/ArmNN-ExecutionProvider.md) (*preview*)|
-||[AMD ROCm](../execution-providers/ROCm-ExecutionProvider.md) (*preview*)|[CoreML](../execution-providers/CoreML-ExecutionProvider.md) (*preview*)|
+|[Intel OpenVINO](../execution-providers/OpenVINO-ExecutionProvider.md)|[AMD MIGraphX](../execution-providers/community-maintained/MIGraphX-ExecutionProvider.md) |[ARM-NN](../execution-providers/community-maintained/ArmNN-ExecutionProvider.md) (*preview*)|
```
>community-maintained [](http://example.com/codeflow?start=110&length=20)

execution-providers ?

TedThemistokleous replied:
The file for this is one level below in the community-maintained folder. I can move it up a level

ytaous commented:
pls also click thru new links you created to make sure it doesn't lead to invalid pages, thx.

---
In reply to: [1369935611](https://github.com/microsoft/onnxruntime/pull/13813#issuecomment-1369935611) [](http://example.com/codeflow?ancestors=1369935611,1369935611)

ytaous reviewed (approved):
pls also wait for review from Peixuan, thx.

TedThemistokleous commented:
@ytaous  I'd like to hold off for another day or so, as we've just recently discovered issue #14126 

I think there's a change that got in to mainline just before the holiday break that's broken builds, and it looks like there's a hole in your CI as tests are passing

ytaous commented:
> @ytaous I'd like to hold off for another day or so, as we've just recently discovered issue #14126
> 
> I think there's a change that got in to mainline just before the holiday break that's broken builds, and it looks like there's a hole in your CI as tests are passing
> 
> It appears a change #13495 is what's caused this

@cloudhan

PeixuanZuo commented:
> @ytaous I'd like to hold off for another day or so, as we've just recently discovered issue #14126
> 
> I think there's a change that got in to mainline just before the holiday break that's broken builds, and it looks like there's a hole in your CI as tests are passing
> 
> It appears a change #13495 is what's caused this

Can we continue this PR if the issue is fixed?

TedThemistokleous commented:
> > @ytaous I'd like to hold off for another day or so, as we've just recently discovered issue #14126
> > I think there's a change that got in to mainline just before the holiday break that's broken builds, and it looks like there's a hole in your CI as tests are passing
> > It appears a change #13495 is what's caused this
> 
> Can we continue this PR if the issue is fixed?

Issue is fixed and confirmed builds are running on our end. I'm finishing up CI and have seen things pass of main branch from last week. This should be good to add in.

ytaous commented:
@faxu

faxu commented on docs/execution-providers/MIGraphX-ExecutionProvider.md:
```diff
@@ -0,0 +1,77 @@
+---
```
There is already a page for this EP here: https://onnxruntime.ai/docs/execution-providers/community-maintained/MIGraphX-ExecutionProvider.html

Can/should it be consolidated?

TedThemistokleous replied:
I added this to mirror what was done for the ROCm EP documentation.  Should we be removing the other then is what you're saying?

TedThemistokleous replied:
Consolidated info to this document page instead and removed the one under community-maintained.

faxu commented:
@TedThemistokleous Could you stage this page for preview to more easily confirm formatting is correct? See instructions here: https://github.com/microsoft/onnxruntime/wiki/Update-ONNX-Runtime-Docs#preview-change

faxu reviewed (changes requested):
Please address comment for consolidation of the EP page and stage a preview of the docs.

TedThemistokleous commented:
@faxu update once your CI found a broken link. This should be good to go now

faxu commented:
@TedThemistokleous Please stage this for preview. See instructions here: https://github.com/microsoft/onnxruntime/wiki/Update-ONNX-Runtime-Docs#preview-change

TedThemistokleous commented:
> @TedThemistokleous Please stage this for preview. See instructions here: https://github.com/microsoft/onnxruntime/wiki/Update-ONNX-Runtime-Docs#preview-change

Hi @Faxu I've done this here's the link: https://tedthemistokleous.github.io/onnxruntime/ 

Oddly I never got an email when this completed. Please let me know if there's anything else required on my end. Thank you for your feedback!

faxu commented on docs/execution-providers/MIGraphX-ExecutionProvider.md:
```diff
@@ -1,54 +1,88 @@
 ## Contents
 {: .no_toc }
 
-* TOC placeholder
```
Can you add lines 18+19 back to display the table of contents for the page?

TedThemistokleous replied:
Sure thing.

TedThemistokleous replied:
Done

faxu commented on docs/install/index.md:
```diff
@@ -244,12 +244,8 @@ The _location_ needs to be specified for any specific version other than the def
 
 ||Official build (location)|Nightly build (location)|
 |---|---|---|
-|PyTorch 1.8.1 (CUDA 10.2)|[**onnxruntime_stable_torch181.cu102**](https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_stable_torch181.cu102.html)|[onnxruntime_nightly_torch181.cu102](https://onnxruntimepackages.z14.web.core.windows.net/onnxruntime_nightly_torch181.cu102.html)|
```
Is this intended to remove the CUDA builds? @ytaous / @PeixuanZuo

faxu commented on index.html:
```diff
@@ -238,9 +238,9 @@ <h3 id="selectHardwareAcceleration">Hardware Acceleration</h3>
-															<span>MIGraphX (Preview)</span></div>	
+															<span>MIGraphX </span></div>
 														<div class="col-lg-3 col-md-3 r-option version" role="option" tabindex="-1" aria-selected="false" id="ROCm">
-															<span>ROCm (Preview)</span></div>		
```
Our criteria for removing "preview" is that it has shipped in a production scenario. Can you confirm this is the case? 
Also, if removing preview, please shift the order of these up in the matrix. The ordering used is: alphabetical, with all non-preview before preview providers

TedThemistokleous replied:
Will do. Let me adjust this

It's my understating that this is in production as this is used and tested with onnxruntime in mind

faxu commented on docs/execution-providers/index.md:
```diff
@@ -31,8 +31,8 @@ ONNX Runtime supports many different execution providers today. Some of the EPs
 |[TVM](../execution-providers/community-maintained/TVM-ExecutionProvider.md) (*preview*)|[DirectML](../execution-providers/DirectML-ExecutionProvider.md)|[Android Neural Networks API](../execution-providers/NNAPI-ExecutionProvider.md)|[Huawei CANN](../execution-providers/community-maintained/CANN-ExecutionProvider.md) (*preview*)|
-|[Intel OpenVINO](../execution-providers/OpenVINO-ExecutionProvider.md)|[AMD MIGraphX](../execution-providers/community-maintained/MIGraphX-ExecutionProvider.md) (*preview*)|[ARM-NN](../execution-providers/community-maintained/ArmNN-ExecutionProvider.md) (*preview*)|
-||[AMD ROCm](../execution-providers/ROCm-ExecutionProvider.md) (*preview*)|[CoreML](../execution-providers/CoreML-ExecutionProvider.md) (*preview*)|
+|[Intel OpenVINO](../execution-providers/OpenVINO-ExecutionProvider.md)|[AMD MIGraphX](../execution-providers/MIGraphX-ExecutionProvider.md)|[ARM-NN](../execution-providers/community-maintained/ArmNN-ExecutionProvider.md) (*preview*)|
```
Same comment about "preview" as on the main matrix page. 
Also, could you help fix the position of "XNNPACK" in the CPU column? There's a 3-row gap. (not due to your change, but just noticed it)

TedThemistokleous replied:
Sure let me add  this in

TedThemistokleous replied:
Done let me know if it looks good

faxu replied:
It looks like the XNNPACK originally listed in the IoT/Edge/Mobile column got shifted accidentally. Can you fix?

TedThemistokleous replied:
Updated

faxu commented:
Please also fix conflicts on the branch, thanks!

TedThemistokleous commented:
@Faxu I don't seem to have write access to this repo to be able to fix said conflicts, nor can I see them when l take a look at the code in the PR on github.

faxu commented:
> ~@faxu I don't seem to have write access to this repo to be able to fix said conflicts, nor can I see them when l take a look at the code in the PR on github.~
> 
> @faxu, fixed changes and merged the head of this to fix conflicts. Let me know if you have any additional issues or changes.

Thanks @TedThemistokleous! Just one more fix on the execution provider index page, otherwise looks good!

TedThemistokleous commented:
> 

Done @Faxu. Let me know if you need anything else.

TedThemistokleous commented:
Hi @faxu  we're still not seeing these changes reflected on the homepage. Is there something wrong with these changes?

PeixuanZuo commented:
> Hi @faxu @ytaous @PeixuanZuo is there something wrong with these changes? It looks like this was somehow backed out as the site doesn't reflect these changes anymore.

There are some changes after this PR and lead to unchanged on https://onnxruntime.ai/. I will fix it.
